x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -192,16 +192,21 @@ private static Attribute resolveAgainstList(UnresolvedAttribute u, Collection<At
         }
 
         if (matches.size() == 1) {
-            return handleSpecialFields(u, matches.get(0), allowCompound);
+            // only add the location if the match is univocal; b/c otherwise adding the location will overwrite any preexisting one
+            return handleSpecialFields(u, matches.get(0).withLocation(u.source()), allowCompound);
```

Comment:
> Why removing it from here and adding it only if matches.size() == 1?

Attribute#`withLocation()` will "overwrite" any pre-existing location, which is not desirable if the existing locations need to be kept in order to be reported in an error message.